manifest/manifest_test.go

```diff
@@ -0,0 +1,114 @@
+package manifest_test
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/filecoin-project/go-f3/certs"
+	"github.com/filecoin-project/go-f3/gpbft"
+	"github.com/filecoin-project/go-f3/manifest"
+	"github.com/stretchr/testify/require"
+)
+
+var baseManifest manifest.Manifest = manifest.Manifest{
+	Sequence:       0,
+	BootstrapEpoch: 10,
+	ReBootstrap:    true,
+	NetworkName:    gpbft.NetworkName("test"),
+	InitialPowerTable: []gpbft.PowerEntry{
+		{
+			ID:     0,
+			PubKey: gpbft.PubKey{0},
+			Power:  big.NewInt(1),
+		},
+		{
+			ID:     1,
+			PubKey: gpbft.PubKey{1},
+			Power:  big.NewInt(1),
+		},
+	},
+	PowerUpdate: []certs.PowerTableDelta{
+		{
+			ParticipantID: 2,
+			PowerDelta:    big.NewInt(1),
+			SigningKey:    gpbft.PubKey{0},
+		},
+		{
+			ParticipantID: 3,
+			PowerDelta:    big.NewInt(1),
+			SigningKey:    gpbft.PubKey{1},
+		},
+	},
+	GpbftConfig: &manifest.GpbftConfig{
+		Delta:                10,
+		DeltaBackOffExponent: 0.2,
+		MaxLookaheadRounds:   10,
+	},
+	EcConfig: &manifest.EcConfig{
+		ECFinality:       900,
+		CommiteeLookback: 5,
+		ECDelay:          30 * time.Second,
+
+		ECPeriod: 30 * time.Second,
+	},
+}
+
+func TestManifestSerialiation(t *testing.T) {
+	b, err := baseManifest.Marshal()
+	require.NoError(t, err)
+
+	var m2 manifest.Manifest
+	fmt.Println(string(b))
+
+	err = m2.Unmarshal(bytes.NewReader(b))
+	require.NoError(t, err)
+	require.Equal(t, baseManifest, m2)
+}
+
+func TestManifestVersion(t *testing.T) {
+	m := baseManifest
+	v1, err := m.Version()
+	require.NoError(t, err)
+	v2, err := baseManifest.Version()
+	require.NoError(t, err)
+	require.Equal(t, v1, v2)
+
+	m.Delta = 1
+	m.NetworkName = "test2"
+	v1, err = m.Version()
+	require.NoError(t, err)
+	require.NotEqual(t, v1, v2)
+
+}
+
+func TestNetworkName(t *testing.T) {
```

Comment:
`TestManifest_NetworkName` ? This is a convention that so far has worked well for me. Ditto for the rest of the tests. At a very least i would include `Manifest` in the test name, since that's what it is testing.